openshift/distro/ubi/9/Dockerfile

```diff
@@ -28,80 +28,53 @@
 
 FROM registry.access.redhat.com/ubi9/ubi-minimal
 
-ENV HOME=/home
-WORKDIR "${HOME}/app"
-
 ARG RUBY_VERSION="3.3"
-
-RUN microdnf module enable -y "ruby:${RUBY_VERSION}" \
- && chown -R 1001:1001 "${HOME}"
-
 ARG BUILD_DEPS="tar make file findutils git patch gcc automake autoconf libtool redhat-rpm-config openssl-devel ruby-devel"
 
-RUN microdnf install -y --nodocs ${BUILD_DEPS}
-
-# Bundler should be kept as-is for productisation
-ARG BUNDLE_VERSION_MATCH=true
-# If the above is false
-ARG BUNDLE_VERSION="2.3.5"
-
-ARG BUNDLE_GEMFILE=Gemfile
-ARG BUNDLE_WITHOUT=development:test
-
-COPY --chown=1001:1001 ${BUNDLE_GEMFILE}.lock "${HOME}/app/"
-
-RUN mkdir -p "${HOME}/.gem/bin" \
- && echo "gem: --bindir ~/.gem/bin --install-dir /usr/share/gems" > "${HOME}/.gemrc"
-
-RUN gem uninstall --executables bundler \
- && BUNDLED_WITH=$(cat ${BUNDLE_GEMFILE}.lock | \
-      grep -A 1 "^BUNDLED WITH$" | tail -n 1 | sed -e 's/\s//g') \
- && if test "${BUNDLE_VERSION_MATCH}x" = "truex"; then \
-      gem install -N bundler --version "${BUNDLED_WITH}"; \
-    else \
-      gem install -N bundler --version "${BUNDLE_VERSION}"; \
-    fi \
- && echo Using $(bundle --version), originally bundled with ${BUNDLED_WITH} \
- && bundle config --local silence_root_warning 1 \
- && bundle config --local disable_shared_gems 1 \
- && bundle config --local without ${BUNDLE_WITHOUT} \
- && bundle config --local gemfile ${BUNDLE_GEMFILE}
+RUN microdnf module enable -y "ruby:${RUBY_VERSION}" \
+ && microdnf install -y --nodocs ${BUILD_DEPS}
 
 # Copy sources
-COPY --chown=1001:1001 ./ "${HOME}/app/"
+WORKDIR /opt/source
+COPY ./ ./
 
-# Builds a clean source tree and deploys it with Bundler.
-# Sets the right configuration and permissions.
+# setup system
 RUN cp -n openshift/3scale_backend.conf /etc/ \
- && chmod 644 /etc/3scale_backend.conf \
- && BACKEND_VERSION=$(gem build apisonator.gemspec | \
-      sed -n -e 's/^\s*Version\:\s*\([^[:space:]]*\)$/\1/p') \
- && gem unpack "apisonator-${BACKEND_VERSION}.gem" --target=/opt/ruby \
- && cd "/opt/ruby/apisonator-${BACKEND_VERSION}" \
- && cp -a ${HOME}/app/.bundle "/opt/ruby/apisonator-${BACKEND_VERSION}/" \
- && bundle config --local path "/opt/ruby/apisonator-${BACKEND_VERSION}/" \
- && bundle install --jobs $(grep -c processor /proc/cpuinfo) \
- && ln -s ${PWD} /opt/app \
- && cp ${HOME}/app/openshift/config/puma.rb ./config/ \
- && cp -n ${HOME}/app/openshift/backend-cron /usr/local/sbin/backend-cron \
- && cp -n ${HOME}/app/openshift/entrypoint.sh ./ \
- && rm -rf ${HOME}/app \
+ && chmod 660 /etc/3scale_backend.conf \
```

Comment:
Permissions where `644` before and are `660` now, I think they could be `640` or even `440` since we don't need to write there. Just a nitpick, not really important, though.

Reply:
if the config file is writable, then it would be possible to run the container and modify config before startup for debugging purposes. I think this might be valuable for debugging. Also I don't see any downsides in having it writable.